src/main/java/me/magnet/consultant/Consultant.java

```diff
@@ -316,8 +327,91 @@ private void init() {
 			Thread.currentThread().interrupt();
 		}
 		catch (ExecutionException e) {
-			throw new RuntimeException(e.getCause());
+			throw new ConsultantException(e.getCause());
+		}
+	}
+
+	public void registerService(int port) {
+		String url = host + "/v1/agent/service/register";
+
+		try {
+			String serviceId = null;
+			String serviceName = id.getServiceName();
+			Check check = new Check("http://" + LOCALHOST + ":" + port + "/_health", 10);
+			ServiceRegistration registration = new ServiceRegistration(serviceId, serviceName, LOCALHOST, port, check);
+			String serialized = mapper.writeValueAsString(registration);
+
+			HttpPut request = new HttpPut(url);
+			request.setEntity(new StringEntity(serialized));
+			request.setHeader("User-Agent", "Consultant");
+			try (CloseableHttpResponse response = http.execute(request)) {
+				int statusCode = response.getStatusLine().getStatusCode();
+				if (statusCode >= 200 && statusCode < 300) {
+					return;
+				}
```

Comment:
`3xx` status codes are just redirects, but there's a good chance that the http client doesn't automatically resolve those. Wouldn't we want to do that anyway?